README.md

```diff
@@ -14,31 +15,38 @@ Via Composer
 $ composer require slim/csrf
 ```
 
-Requires Slim 3.0.0 or newer.
+Requires Slim 4.0.0 or newer.
 
 ## Usage
 
-In most cases you want to register Slim\Csrf for all routes, however,
-as it is middleware, you can also register it for a subset of routes.
-
+In most cases you want to register Slim\Csrf for all routes, however, as it is middleware, you can also register it for a subset of routes.
 
 ### Register for all routes
 
 ```php
+use DI\Container;
+use Slim\Csrf\Guard;
+use Slim\Factory\AppFactory;
+
+require __DIR__ . '/vendor/autoload.php';
+
 // Start PHP session
 session_start();
 
-$app = new \Slim\App();
+// Create Container
+$container = new Container();
 
-// Register with container
-$container = $app->getContainer();
-$container['csrf'] = function ($c) {
-    return new \Slim\Csrf\Guard;
-};
+// Create App
```

Comment:
We need to inform the app factory about the container 😃 before creating the app.
```php
AppFactory::setContainer($container);
```